File: task-03/HelloWorld.hs
isPrime :: Integer -> Bool
isPrime n
  | n < 2     = False
  | n == 2    = True
  | even n    = False
  | otherwise = not $ any (\x -> n `mod` x == 0) [3,5..isqrt n]
  where
    isqrt = floor . sqrt . fromIntegral

primesUpToN :: Integer -> [Integer]
primesUpToN n = filter isPrime [2..n]

main :: IO ()
main = do
  putStrLn "Enter a positive integer (n): "
  userInput <- getLine
  let n = read userInput :: Integer
  if n >= 2
    then putStrLn $ "Prime numbers up to " ++ show n ++ ": " ++ show (primesUpToN n)
    else putStrLn "Please enter a positive integer greater than or equal to 2."

File: task-03/HelloWorld.ex
defmodule Prime do
  def nth_prime(n), do: prime(n, 2)
def is_prime(x), do: (2..x |> Enum.filter(fn a -> rem(x, a) == 0 end) |> length()) == 1
def prime(n, n), do: []
def prime(n, acc) do
  case is_prime(acc) do
    true -> [acc | prime(n, acc + 1)]
    false -> prime(n, acc + 1)
  end
end 
  
  def start do
    {n, _} = IO.gets("") |> Integer.parse

    i = nth_prime(n)
    IO.inspect i
  end
  
end

Prime.start()